crates/lib/src/qpu/rewrite_arithmetic.rs

```diff
@@ -50,6 +50,9 @@ use crate::executable::Parameters;
 /// distinct parameter from theta. Note that the value of `__SUBST[0]` will actually be
 /// `theta * 1.5 / 2π`.
 pub fn rewrite_arithmetic(program: Program) -> Result<(Program, Substitutions), Error> {
+    #[cfg(feature = "tracing")]
+    tracing::debug!("rewriting arithmetic");
```

Comment:
Seems like `tracing::instrument` might be the right way to go for some of these, too? I'm not sure if I have a good rule of thumb for when we'd want: 

`tracing::instrument`

vs

`tracing::instrument(ret, ...)`

vs

event with `tracing::debug` etc

but a rough guess would be "call to external library or service, or prior to trying something likely to fail". However, even "likely to fail" could be `tracing::instrument(err, ...)`, right?

`tracing-error` has been _fantastic_ on other projects to surface spans only in the case of errors, but that might be a bit much to add here.

Reply:
Haven't used `tracing-error` before, so I'll at least look at it.

I don't really have a good rule of thumb on these either. In this case, rewriting arithmetic seemed like something worth surfacing as a logged event.